discovery-infra/download_logs.py

```diff
@@ -80,9 +81,11 @@ def get_clusters(client, all_cluster):
     return client.clusters_list()
 
 
-def should_download_logs(cluster: dict):
-    return cluster['status'] in [ClusterStatus.ERROR]
-
+def should_download_logs(cluster_events: list):
```

Comment:
nit: Using `list` directly as a type annotation is a Python 3.9+ feature, for older versions `from typing import List` must be used. Also please specify the list type with `[]`